libvips/colour/CMYK2XYZ.c

```diff
@@ -150,20 +150,57 @@ vips_CMYK2XYZ_build( VipsObject *object )
 	return( 0 );
 }
 
+#else
+
+typedef VipsColourCode VipsCMYK2XYZ;
+typedef VipsColourCodeClass VipsCMYK2XYZClass;
+
+G_DEFINE_TYPE(VipsCMYK2XYZ, vips_CMYK2XYZ, VIPS_TYPE_COLOUR_CODE);
+
+#define SCALE (VIPS_D65_Y0)
+
+void
+vips_CMYK2XYZ_line( VipsColour *colour, VipsPel *out, VipsPel **in, int width )
+{
+    unsigned char *p = (unsigned char *) in[0];
+    float *q = (float *) out;
+
+    int i;
+
+    for (i = 0; i < width; i++) {
+        float r = 255.0 - ((1.0 / 255.0) * p[0] * (255.0 - p[3]) + p[3]);
+        float g = 255.0 - ((1.0 / 255.0) * p[1] * (255.0 - p[3]) + p[3]);
+        float b = 255.0 - ((1.0 / 255.0) * p[2] * (255.0 - p[3]) + p[3]);
+        
+        float x, y, z;
+        
+        vips_col_scRGB2XYZ(r / 255.0, g / 255.0, b / 255.0, &x, &y, &z);
+        q[0] = x;
+        q[1] = y;
+        q[2] = z;
+
+        p += 4;
+        q += 3;
+    }
+}
+
+#endif /*HAVE_LCMS2*/
```

Comment:
I don't like mixing up the two `CMYK2XYZ` operations with `ifdefs`. It would be very easy to make a mistake in the future.

I think it would be better to have two completely separate implementations with a single ifdef to flip between them. Something like:

```C
// prelude

#ifdef HAVE_LCMS2

// implementation based on icc_import

#else /*!HAVE_LCMS2*/

// implementation based on ColourCode with your own maths

#endif /*HAVE_LCMS2*/

/**
 * doc comment
 */
int
vips_CMYK2XYZ( VipsImage *in, VipsImage **out, ... )
{
// etc.
}
```

Reply:
Yes, it more readable and better for futur dev